src/PaymentGateways/Gateways/Stripe/Actions/UpdateStripeFormBuilderSettingsMeta.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Give\PaymentGateways\Gateways\Stripe\Actions;
+
+use Give\DonationForms\Models\DonationForm;
+
+/**
+ * Update Stripe settings on backwards compatible form meta.
+ *
+ * @since 3.0.0
+ */
+class UpdateStripeFormBuilderSettingsMeta
+{
+    /**
+     * @since 3.0.0
+     * @param  DonationForm  $form
+     */
+    public function __invoke(DonationForm $form)
+    {
+        $paymentGatewaysBlock = $form->blocks->findByName('givewp/payment-gateways');
+        if (!$paymentGatewaysBlock) {
+            give()->form_meta->delete_meta($form->id, 'give_stripe_per_form_accounts');
+            give()->form_meta->delete_meta($form->id, '_give_stripe_default_account');
+
+            return;
+        }
+
+        [
+            'stripeUseGlobalDefault' => $stripeUseGlobalDefault,
+            'stripeAccountId' => $stripeAccountId
+        ] = $paymentGatewaysBlock->getAttributes() + ['stripeUseGlobalDefault' => true, 'stripeAccountId' => ''];
+
+        if (is_bool($stripeUseGlobalDefault)) {
+            give()->form_meta->update_meta($form->id, "give_stripe_per_form_accounts", $stripeUseGlobalDefault ? 'disabled' : 'enabled');
+        }
+
+        if (is_string($stripeAccountId)) {
```

Comment:
We should probably check if `$stripeUseGlobalDefault` is true as well.  Stripe uses prefixes for their ID's so you could even make sure this string includes `acct_`

Reply:
Sure.

Reply:
Resolved: https://github.com/impress-org/givewp/pull/6903/commits/404623fb809d02ea78db4d442f93429001398dbb